modules/plan/src/main/PlanApi.scala

```diff
@@ -181,6 +174,24 @@ final class PlanApi(
         }
     }
 
+  def onCompletedSession(completedSession: StripeCompletedSession): Funit =
+    customerIdPatron(completedSession.customer) flatMap {
+      case None =>
+        logger.warn(s"Completed Session of unknown patron $completedSession")
+        funit
+      case Some(patron) =>
+        userRepo byId patron.userId orFail s"Missing user for $patron" flatMap { user =>
+          saveStripePatron(
+            user,
+            completedSession.customer,
+            completedSession.mode match {
```

Comment:
or `if (completedSession.mode == "subscription") Freq.Monthly else Freq.Onetime` ;)

Reply:
Yes, this is better in this case.